### PR TITLE
feat(slider): add support for mirroring slider

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #720

## Summary

This adds a `mirrored` property to `calcite-slider` (default false). When set to true, it will invert the slider unless there is a histogram associated. In this case, `mirrored` has no effect. This decision is based on https://material.io/design/usability/bidirectionality.html#mirroring-layout, talking to some teams that work w/ data visualization + sliders and some good ol' fashioned googling.

**Extras**

* sorts props alphabetically
* fixed casing of an internal type
* refactored method arguments/vars to make them value-agnostic - these are now left/right-based
* adds local vars to store bounding client rects to avoid duplicate computations
* adds another column for testing mirrored along with some edge-positioning cases
    Screen Shot 2021-07-07 at 2 33 32 PM

### CODE REVIEW

File: src/components/calcite-slider/calcite-slider.tsx [hunk 1]
```diff
@@ -1158,33 +1178,32 @@ export class CalciteSlider {
    */
   private getHostOffset(leftBounds: number, rightBounds: number): number {
     const hostBounds = this.el.getBoundingClientRect();
-    if (leftBounds + 7 < hostBounds.left) {
-      const offset = hostBounds.left - leftBounds - 7;
-      return offset;
+    const buffer = 7;
```
Comment (driskull) [inline src/components/calcite-slider/calcite-slider.tsx:1181]: nitpick. Should we store this in the resources file?

File: src/components/calcite-slider/calcite-slider.tsx [hunk 2]
```diff
@@ -41,57 +42,64 @@ export class CalciteSlider {
...
+
+  @Watch("histogram") histogramWatcher(newHistogram: DataSeries): void {
+    this.hasHistogram = !!newHistogram;
+  }
+
+  /** Label handles with their numeric value */
+  @Prop({ reflect: true }) labelHandles?: boolean;
+
+  /** Label tick marks with their numeric value. */
+  @Prop({ reflect: true }) labelTicks?: boolean;
```
Comment (driskull) [inline src/components/calcite-slider/calcite-slider.tsx:59]: nitpick: Do we have guidelines for when boolean props should be initialized to false vs undefined?

For example, hasHistagram = false but labelTicks = undefined. mirrored = false;

Its not a big deal just a consistency thing :)

File: src/utils/dom.ts [hunk 3]
```diff
@@ -215,3 +215,12 @@ export function setRequestedIcon(
     return iconObject[matchedValue];
   }
 }
+
+export function intersects(rect1: DOMRect, rect2: DOMRect): boolean {
```
Comment (driskull) [inline src/utils/dom.ts:219]: 👏

**Review by driskull — APPROVED** [review]
LGTM!

Nice cleanup as well!

File: src/components/calcite-slider/calcite-slider.tsx (hunk 2 above)
Comment (driskull) [reply src/components/calcite-slider/calcite-slider.tsx:59]: I know we have the same inconsistencies with strings. Some are undefined by default and others are "";

Comment (jcfranco) [reply src/components/calcite-slider/calcite-slider.tsx:59]: We don't have any guidelines at the moment for defaults. Whenever possible, I think we should initialize to a sensible value (similar to native elements). This use case is weird to me if we don't:

```ts
const slider = document.createElement("calcite-slider");
console.log(slider.labelTicks); // undefined even though the doc types it as boolean
```

There could be some exceptions. For example, if we treat `intl` strings as overrides instead of defaults we could skip initializing to keep the defaults internal and to also promote reuse between components (e.g., [`calcite-time-picker#intlHour`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-time-picker/calcite-time-picker.tsx#L59) vs. [`calcite-input-time-picker#intlHour`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-input-time-picker/calcite-input-time-picker.tsx#L49) )

File: src/utils/dom.ts (hunk 3 above)
Comment (eriklharper) [reply src/utils/dom.ts:219]: this is awesome!  Now it can be used for more than just the x direction.

**Review by eriklharper — APPROVED** [review]
Excellent work!  Love the more semantic variable names that apply to the left/right versus min/max.  Makes the code so much easier to read and understand.

**Comment by bstifle** [comment]
This has some weird problems with keyboard nav. Left goes right and right goes left... cant those stay the same for RTL? west isnt east in arabic ;)

**Comment by jcfranco** [comment]
@bstifle Fixed left/right keyboard issue.
